conan/tools/gnu/autotoolstoolchain.py

```diff
@@ -120,11 +133,99 @@ def defines(self):
         ret = [self.ndebug, self.gcc_cxx11_abi] + conf_flags + self.extra_defines
         return self._filter_list_empty_fields(ret)
 
+    def _get_cc(self):
+        return self._exe_env_var_to_unix_path(
+            "CC",
+            "cl" if is_msvc(self._conanfile) else None,
+            ["-nologo"] if is_msvc(self._conanfile) else [],
+            self._compile_wrapper,
+        )
+
+    def _get_cxx(self):
+        return self._exe_env_var_to_unix_path(
+            "CXX",
+            "cl" if is_msvc(self._conanfile) else None,
+            ["-nologo"] if is_msvc(self._conanfile) else [],
+            self._compile_wrapper,
+        )
+
+    def _get_ld(self):
+        return self._exe_env_var_to_unix_path(
+            "LD",
+            "link" if is_msvc(self._conanfile) else None,
+            ["-nologo"] if is_msvc(self._conanfile) else [],
+        )
+
+    def _get_ar(self):
+        return self._exe_env_var_to_unix_path(
+            "AR",
+            "lib" if is_msvc(self._conanfile) else None,
+            None, # do not add any option, ar wrapper may not like this
+            self._ar_wrapper,
+        )
+
+    def _get_nm(self):
+        return self._exe_env_var_to_unix_path(
+            "NM",
+            "dumpbin" if is_msvc(self._conanfile) else None,
+            ["-nologo", "-symbols"] if is_msvc(self._conanfile) else [],
+        )
+
+    def _get_objdump(self):
+        return self._exe_env_var_to_unix_path(
+            "OBJDUMP",
+            ":" if is_msvc(self._conanfile) else None,
+        )
+
+    def _get_ranlib(self):
+        return self._exe_env_var_to_unix_path(
+            "RANLIB",
+            ":" if is_msvc(self._conanfile) else None,
+        )
+
+    def _get_strip(self):
+        return self._exe_env_var_to_unix_path(
+            "STRIP",
+            ":" if is_msvc(self._conanfile) else None,
+        )
+
+    def _exe_env_var_to_unix_path(self, env_var, default=None, extra_options=None, wrapper=None):
+        """
+            Convenient method to convert env vars like CC, CXX or LD to values compatible with autotools.
+            If env var doesn't exist, returns default.
+        """
+        exe = VirtualBuildEnv(self._conanfile).vars().get(env_var)
```

Comment:
I've tested and it's always `None`, regardless of env var I put in my profile.

I've tried with `VirtualBuildEnv` and `conans.tools.get_env`, what's wrong? I see that `MesonToolchain` also tries to get `CC`, `CXX` etc from profile with `VirtualBuildEnv`, does it work actually?

Reply:
I don't have much experience with conan 2.0 (yet), but did you use a `[buildenv]` section in your host profile? See [this link](https://docs.conan.io/en/2.0/tutorial/consuming_packages/cross_building_with_conan.html?highlight=profile#conan-two-profiles-model-build-and-host-profiles).

Reply:
Indeed thanks, it works now.

But now I see two issues:

- if `VirtualBuildEnv` is called after `AutotoolsToolchain` in `generate()`, it overrides CC etc, so it's very important to call `VirtualBuildEnv` first in recipes.
- if there is a space in these paths (which is likely the case for cl, link in Program Files), even `unix_path` doesn't help

Reply:
For the second point, the best I can do is display a warning and suggest to set CC etc with short path

For example on my computer I can set in profile:

```
CC=C:\PROGRA~1\MIB055~1\2022\Community\VC\Tools\MSVC\14.33.31629\bin\Hostx64\x64\cl.exe
```